kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/fromstunner/properties/MultipleInstanceActivityPropertyWriter.java

```
@@ -221,6 +221,8 @@ protected static void addSafe(List<DataOutputAssociation> associations, DataOutp
     }
 
     private Property findPropertyById(String id) {
```

see my comments in FlatVariableScope. Before, if there was an inconsistency in the process an exception was thrown complaining about the missing variable. And hopefully the user could see an error popup, now a NPE will be produced.

---

now if the property is not found by a given id, it is being returned a `null`

---

yes, I could see that a null is returned. But question is, how can this affect to callers of this method. Since a null es returned, the callers of this method will fail by returning a NPE. But before, an exception was thrown at least saying that the expected variable XXX was not found

---

yeah, I see, great point, the callers should be handling the `null` value but they are not, IMO we can skip the property in case it is not found, WDYT? or you think it is better to throw an exception?

---

If I don't remember wrong, in current version, when a variable is missing, an exception is thrown and the xml editor is shown. I believe that this is because the variable might me referenced in different places (e.g. in several UserTask assignments) and thus the process can remain with several inconsistent configurations. On the other hand, if the process is still opened (by skipping the missing variable), and the variable is used, and you do a silly modification, and try to save the process, I believe the on save transformation will also fail because of the missing variable.
So I don't know if to just skip the missing property will work fine.
Probably the best/quickest option is to inform the user about the missing variable, and open the xml editor as it's being done now. (the other alternative is to do a more elaborated research of the impact of still opening the process by skipping the missing variable. I don't know all the use cases that might happen to be honest)
WDYT?

@romartin I'll add Roger to the thread in case he can came up with some other alternative.